### PR TITLE
Don't try to PUT descriptions to non-existant binaries

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3648

# What does this Pull Request do?
Adds a check to stop an attempt to `PUT` a binary description to a binary that doesn't exist

# How should this be tested?

Before PR
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/bob                   
HTTP/1.1 404 Not Found
Date: Fri, 19 Feb 2021 15:50:08 GMT
Set-Cookie: JSESSIONID=node01ggergt4zt1gw1nxk193s7u6cg0.node0; Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 18-Feb-2021 15:50:09 GMT; SameSite=lax
Content-Type: text/turtle;charset=utf-8
Content-Length: 48
Server: Jetty(9.4.34.v20201102)

Error: No OCFL mapping found for info:fedora/bob

> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/bob/fcr:metadata -XPUT
HTTP/1.1 500 Internal Server Error
Set-Cookie: JSESSIONID=node01ksys0g9n7mxs12ow2beuziqp1.node0; Path=/
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 18-Feb-2021 15:50:11 GMT; SameSite=lax
Content-Length: 0
Server: Jetty(9.4.34.v20201102)
```

After this PR
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/bob
HTTP/1.1 404 Not Found
Date: Fri, 19 Feb 2021 15:33:34 GMT
Set-Cookie: JSESSIONID=node0bkyb2qrk8qjs7cw9jfrjivf60.node0; Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 18-Feb-2021 15:33:34 GMT; SameSite=lax
Content-Type: text/turtle;charset=utf-8
Content-Length: 48
Server: Jetty(9.4.34.v20201102)

Error: No OCFL mapping found for info:fedora/bob

> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/bob/fcr:metadata -XPUT
HTTP/1.1 404 Not Found
Date: Fri, 19 Feb 2021 16:06:48 GMT
Set-Cookie: JSESSIONID=node0abt77vjok7n41nppcclaq2xcd0.node0; Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 18-Feb-2021 16:06:48 GMT; SameSite=lax
Content-Type: text/plain
Content-Length: 36
Server: Jetty(9.4.34.v20201102)

Error: Binary at path /bob not found
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 1]
```diff
@@ -420,6 +420,11 @@ public Response createOrReplaceObjectRdf(
                 throw new GhostNodeException("Resource path " + externalPath() + " is an immutable resource.");
             }
 
+            if (!resourceExists && fedoraId.isDescription()) {
+                // Can't PUT a description to a non-existant binary.
+                throw new PathNotFoundException("Binary at path " + fedoraId.asBaseId().getFullIdPath() + " not found");
```
Comment (pwinckles) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java:425]: This prints the following when you try to PUT to `/rest/fcr:metdata`:

> Error: Binary at path  not found

Maybe we should put a leading `/` or single quotes around it?

**Review by pwinckles — COMMENTED** [review]
Works for me. One minor comment about the exception text

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java (hunk 1 above)
Comment (whikloj) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java:425]: Ok, I can add some simple handling for a repository root base ID.

**Review by pwinckles — APPROVED** [review]
👍

**Pull request merged**